.gitignore

```diff
@@ -0,0 +1,165 @@
+arsenal/datastructures/heap/heap.c
```

Comment:
I copied this .gitignore from Ben's `genlm-backend`. Obviously overkill, I'm happy to trim it down if you like!